src/libcommonserver/log/log.cpp

```diff
@@ -109,4 +115,393 @@ Log::Log(const log4cplus::tstring &filePath) {
     LOG_INFO(_logger, "Logger initialization done");
 }
 
+bool Log::getLogEstimatedSize(uint64_t &size, IoError &ioError) {
+    const SyncPath logPath = _filePath.parent_path();
+    ioError = IoErrorSuccess;
+    for (int i = 0; i < 2; i++) {  // Retry once in case a log file is archived/created during the first iteration
+        bool tooDeep = false;
+        bool result = IoHelper::getDirectorySize(logPath, size, 0, tooDeep, ioError);
+        if (result && ioError == IoErrorSuccess) {
+            return true;
+        }
+    }
+    return false;
+}
+
+SyncPath Log::getLogFilePath() const {
+    return _filePath;
+}
+
+ExitCode Log::generateLogsSupportArchive(bool includeOldLogs, const SyncPath &outputPath, const SyncPath &archiveName,
+                                         ExitCause &exitCause, std::function<void(int)> progressCallback) {
+    // Get the log directory path
+    const SyncPath logPath = _filePath.parent_path();
+    const SyncPath tempDirectory =
+        logPath / "send_log_directory_temp" / ("tempLogArchive_" + CommonUtility::generateRandomStringAlphaNum(10));
```

Comment:
Define this string as static const variable : `send_log_directory_temp`

Reply:
As discussed, the string duplication is removed in commits: [c147e5d](https://github.com/Infomaniak/desktop-kDrive/pull/100/commits/c147e5daea61a0f170699d10a6600b9f8be48ce9) & [bc206e5](https://github.com/Infomaniak/desktop-kDrive/pull/100/commits/bc206e5dda3bc669ed62d0536330f4bfe08f4dd9)